### PR TITLE
0-Sixty: Splits kb config into base and underglow

### PR DESCRIPTION
**Pull request by vinamarora8**
## Description

Splits 0_sixty keyboard config into two: base (old) and underglow (new). This is done to support a mod where the caps-lock LED pin is used for a WS2182B LED strip instead. 

Since most of the config remains common between the two, I have chosen to not create a new keyboard folder, but split this one into 2 'sub-keyboard' folders.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/0_sixty/underglow/rules.mk [hunk 1]
```diff
@@ -0,0 +1,2 @@
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+EXTRA_FLAGS += -flto # Link-time opt to reduce f/w size
```
Comment (drashna) [inline keyboards/0_sixty/underglow/rules.mk:2]:
```suggestion
LTO_ENABLE = yes
```

File: keyboards/0_sixty/rules.mk [hunk 2]
```diff
@@ -19,3 +19,5 @@ BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
 RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
 BLUETOOTH_ENABLE = no       # Enable Bluetooth
 AUDIO_ENABLE = no           # Audio output
+
+DEFAULT_FOLDER = 0_sixty/base
```
Comment (drashna) [inline keyboards/0_sixty/rules.mk:23]: Honestly, no need to use "base".  In fact, I'd say that doing so makes things more complicated.  You can compile just `0_sixty` and get the base, and then `0_sixty/underglow` if you need. 

You can take a look at the ergodox ez, and the glow and shine revisions for an example of this.

Comment (vinamarora8) [reply keyboards/0_sixty/rules.mk:23]: This looks like a better way of handling it. THank

**Comment by vinamarora8** [comment]
@drashna, I've made all the changes according to your suggest. This way it is much much simpler, you were right.

File: keyboards/0_sixty/underglow/config.h [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+#define PRODUCT         0-Sixty-underglow
+
+/* no caps lock with underglow */
+#undef LED_CAPS_LOCK_PIN
+
+/* underglow */
+#define RGB_DI_PIN B6
+#define RGBLED_NUM 24
+#define RGBLIGHT_LIMIT_VAL 185 // limit max I to 400mA 
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/0_sixty/underglow/config.h:34]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**